tests/Resources/ti.geolocation.test.js

```diff
@@ -303,16 +308,263 @@ describe.windowsBroken('Titanium.Geolocation', () => {
 			it('is a Function', () => {
 				should(Ti.Geolocation).have.a.property('getCurrentPosition').which.is.a.Function();
 			});
+
+			// FIXME: On local Apple Silicon (M1) emulator these fail because passive/network providers are not available
+			it('works via callback argument', function (finish) {
+				this.timeout(6e4); // 60 sec
+
+				// can't get permissions on macOS or actual iOS devices, since it prompts
+				if ((isMacOS || isIOSDevice) && !Ti.Geolocation.hasLocationPermissions(permission)) {
+					return finish(); // FIXME: How can we limit to ios only, and skip on macos?
+				}
+
+				Ti.Geolocation.accuracy = Ti.Geolocation.ACCURACY_HIGH;
+
+				function testCurrentPosition() {
+					console.info('Testing getCurrentPosition() w/ callback...');
+					Ti.Geolocation.getCurrentPosition(function (data) {
+						console.info('getCurrentPosition() callback returned!');
+						try {
+							should(data).have.property('success').which.is.true();
+							should(data).have.property('code').which.eql(0);
+							should(data.coords).be.an.Object();
+							should(data.coords).have.a.property('latitude').which.is.a.Number();
+							should(data.coords).have.a.property('longitude').which.is.a.Number();
+							should(data.coords).have.a.property('altitude').which.is.a.Number();
+							should(data.coords).have.a.property('accuracy').which.is.a.Number();
+							should(data.coords).have.a.property('heading').which.is.a.Number();
+							should(data.coords).have.a.property('speed').which.is.a.Number();
+							should(data.coords).have.a.property('timestamp').which.is.a.Number();
+						} catch (err) {
+							console.log(data);
+							return finish(err);
+						}
+						finish();
+					});
+				}
+
+				if (!Ti.Geolocation.hasLocationPermissions(permission)) {
+					console.info('Requesting location permissions, because we do not have them...');
+					Ti.Geolocation.requestLocationPermissions(permission, function (e) {
+						if (!e.success) {
+							console.log(e);
+							return finish(new Error('Failed to get Geolocation ALWAYS permission: ' + e.error));
+						}
+						testCurrentPosition();
+					});
+				} else {
+					testCurrentPosition();
+				}
+			});
+
+			it('works via Promise return value', function (finish) {
+				this.timeout(6e4); // 60 sec
+
+				// can't get permissions on macOS or actual iOS devices, since it prompts
+				if ((isMacOS || isIOSDevice) && !Ti.Geolocation.hasLocationPermissions(permission)) {
+					return finish(); // FIXME: How can we limit to ios only, and skip on macos?
+				}
+
+				Ti.Geolocation.accuracy = Ti.Geolocation.ACCURACY_HIGH;
+
+				function testCurrentPosition() {
+					console.info('Testing getCurrentPosition() w/ Promise...');
+					const result = Ti.Geolocation.getCurrentPosition();
+					try {
+						result.should.be.a.Promise();
+					} catch (err) {
+						return finish(err);
+					}
+					result.then(data => {
+						console.info('getCurrentPosition() promise resolved!');
+						try {
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.geolocation.test.js;
  Line: 532;
-->

- :warning: tests/Resources/ti.geolocation.test.js line 532 – Each then() should return a value or throw (promise/always-return)